### PR TITLE
Maltiverse URL Validation improvements

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/27090|

## Description

Improves the Maltiverse's integration URL validation and adds UT to this functionality.

### CODE REVIEW

File: integrations/tests/test_maltiverse.py [hunk 1]
```diff
@@ -140,6 +140,21 @@ def test_get_by_sha1():
...
 
 
+@pytest.mark.parametrize('url,expected', [
+    ('http://example.com', True),
+    ('https://example.com:8080', True),
+    ('https://api.maltiverse.com', True),
+    ('http://example.com/<script>', False),
+    ('https://192.168.0.1:invalid', False),
+    ('malformed://example.com', False),
+    ('http://', False)
```
Comment (GGP1) [inline integrations/tests/test_maltiverse.py:150]: Let's add some URLs from the [issue](https://github.com/python/cpython/issues/103848) that reported the CVE to be sure we reject them if there are changed in the future:

```
https://user:some]password[@host.com
https://user:some%5Dpassword%5B@host.com
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

File: integrations/tests/test_maltiverse.py (hunk 1 above)
Comment (RamosFe) [reply integrations/tests/test_maltiverse.py]: Added in 76c13c142be9b4c40c3f97316fef6871ccc554cc.

```console
(env) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh$ PYTHONPATH=/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/:/home/federamos/Documents/Wazuh/Repositories/wazuh/api python3 -m pytest integrations/
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/integrations
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, cov-4.1.0, tavern-1.23.5, anyio-4.1.0, trio-0.8.0
asyncio: mode=auto
collected 131 items                                                                                                                                                                                                                     

integrations/tests/test_maltiverse.py .......................................................................                                                                                                                     [ 54%]
integrations/tests/test_pagerduty.py ..........                                                                                                                                                                                   [ 61%]
integrations/tests/test_shuffle.py ..................                                                                                                                                                                             [ 75%]
integrations/tests/test_slack.py ..........                                                                                                                                                                                       [ 83%]
integrations/tests/test_virustotal.py ......................                                                                                                                                                                      [100%]

========================================================================================================== 131 passed in 2.92s ==========================================================================================================
```

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by nico-stefani — APPROVED** [review]
LGTM

**Pull request merged**